pymie/mie.py

```diff
@@ -251,11 +254,142 @@ def calc_integrated_cross_section(m, x, wavelen_media, theta_range):
 
     # np.trapz does not preserve units, so need to state explicitly that we are
     # in the same units as the integrand
-    integral_par = 2 * np.pi * np.trapz(integrand_par, x=angles)*integrand_par.units
-    integral_perp = 2 * np.pi * np.trapz(integrand_perp, x=angles)*integrand_perp.units
+    integral_par = 2 * np.pi * np.trapz(integrand_par, x=angles.magnitude)#*integrand_par.units new pint preserves units
+    integral_perp = 2 * np.pi * np.trapz(integrand_perp, x=angles.magnitude)#*integrand_perp.units new pint preserves units
 
     # multiply by 1/k**2 to get the dimensional value
     return wavelen_media**2/4/np.pi/np.pi * (integral_par + integral_perp)/2.0
+    
+def calc_energy(radius, n_medium, m, x, nstop,
+           eps1 = DEFAULT_EPS1, eps2 = DEFAULT_EPS2):
```

Comment:
can you add a test?

Reply:
There is already a test that compares the value returned from this function to an example from a paper. Since it uses the same parameters for testing the dwell time calculations, I put them in the same test function. I just renamed the function from test_dwell_time() to test_dwell_time_and_energy() to make it clear what is being tested.